system/jlib/jstats.cpp

```diff
@@ -1005,6 +1005,7 @@ class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
         {
             Statistic * next = Statistic::deserialize(in, version);
             stats.append(*next);
+            next->Release();
```

Comment:
Yes, I coded it this way on purpose. I would like @ghalliday to take a look when he returns - I'm not sure exactly what his intent was.